### PR TITLE
Version updates and subsequent fixes

### PR DESCRIPTION
**Pull request by evanchooly**
I bumped the version of gradle up which brought along checkstyle/findbugs updates.  this PR also cleans up those new failures.  clirr has been nonfunctional for a while so it is also removed in this PR.

### CODE REVIEW

**Comment by jyemin** [comment]
Looks like Travis is failing.

> Could not find method buildingWith() for arguments [xmlReports.enabled] on project ':entityscanner-plug' of type org.gradle.api.Project.

File: build.gradle [hunk 1]
```diff
@@ -25,7 +23,6 @@ buildscript {
     }
 
     dependencies {
-        classpath 'org.kordamp.gradle:clirr-gradle-plugin:0.2.2'
```
Comment (jyemin) [inline build.gradle]: I hate to lose clirr.... What's broken about it and can it be fixed?

File: no-proxy-deps-tests/build.gradle [hunk 2]
```diff
@@ -1,8 +1,16 @@
 sourceSets {
     main {
-        java.outputDir = project.project(":util").sourceSets.test.output.files[0]
+        java.outputDir = project.project(":morphia").sourceSets.test.output.files[0]
```
Comment (jyemin) [inline no-proxy-deps-tests/build.gradle]: What's this change about?

File: build.gradle [hunk 3]
```diff
@@ -141,12 +139,15 @@ configure(subprojects.findAll { it.name != 'util' }) {
     javadoc {
         exclude "**/internal/**"
 
-        dependsOn project(':util').compileJava //We need taglets to be compiled
+        dependsOn project.project(':util').compileJava //We need taglets to be compiled
```
Comment (jyemin) [inline build.gradle]: For my edification... what necessitated this change?  What does project.project mean?

File: build.gradle (hunk 1 above)
Comment (evanchooly) [reply build.gradle]: It hasn't worked for a long, long time.  Keeps giving errors about missing classes and the like.  I've tried again and again to compare to the driver's configuration but haven't been able to spot what's different.

File: build.gradle (hunk 3 above)
Comment (evanchooly) [reply build.gradle]: that's a remnant of some refactoring to the new gradle API.   I'll clean that up in a subsequent PR.  It looks like the docs aren't generating valid links anymore to the mongo docs so i'll have a PR to clean that up soon once I track it down.

**Pull request merged**